roles/kubespray-defaults/tasks/fallback_ips.yml

```diff
@@ -0,0 +1,30 @@
+---
+# Set 127.0.0.1 as fallback IP if we do not have host facts for host
+# ansible_default_ipv4 isn't what you think.
+# Thanks https://medium.com/opsops/ansible-default-ipv4-is-not-what-you-think-edb8ab154b10
+
+- name: Gather ansible_default_ipv4 from all hosts
+  setup:
+    gather_subset: '!all,network'
+    filter: "ansible_default_ipv4"
+  delegate_to: "{{ item }}"
+  delegate_facts: yes
+  with_items: "{{ groups['all'] }}"
+  run_once: yes
```

Comment:
Perhaps adding `ansible_all_ipv4_addresses` for additional fallback when `ansible_default_ipv4` is empty (when the host does not have a default route)?
Also perhaps adding `when: hostvars[item].ansible_default_ipv4 is not defined and hostvars[item]. ansible_all_ipv4_addresses is not defined` to speed up gathering.